system/settings/settings.c

```diff
@@ -249,7 +249,6 @@ static int set_string(FAR setting_t *setting, FAR char *str)
       return -EACCES;
     }
 
-  ASSERT(strlen(str) < CONFIG_SYSTEM_SETTINGS_VALUE_SIZE);
   if (strlen(str) >= CONFIG_SYSTEM_SETTINGS_VALUE_SIZE)
```

Comment:
`strlen(str) != CONFIG_SYSTEM_SETTINGS_VALUE_SIZE` ? :-)

Reply:
Similarly to my comment about key lengths: values can have mixed lengths.